### PR TITLE
REST API

### PR DESCRIPTION
**Pull request by BohdanDudkaTD**
This PR expands the functionality of the `RequestService` by implementation PUT HTTP request. Also, all methods now accept `URL` by which requests should be sent.
Besides that few operations were added to services. Now `UserService` can send requests to `get user info`. `FileService` now can send requests to:
- upload file or folder;
- rename file or folder;
- create folder;
- get sharable URL of the file or folder;
- move file or folder;
- download file or folder;

Result of the ESLint validation:
image

### CODE REVIEW

**Comment by BohdanDudkaTD** [comment]
@dkalinin, PTAL

**Comment by BohdanDudkaTD** [comment]
Fixed issues:
- merge conflict;
- JS docs to match convention;
- the payload was not passed to the getFiles method.

File: app/router/index.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 export class Router {
   /**
    * Sets router`s routing configuration - URL`s hash pattern and corresponding class name.
-   * Create empty array for possible redirections. Add event handler for hashchange event.
+   * Create empty array for possible redirects. Add event handler for hashchange event.
```
Comment (dkalinin) [inline app/router/index.js]: Creates

Comment (dkalinin) [reply app/router/index.js]: Adds event handler...

File: app/router/index.js [hunk 2]
```diff
@@ -7,7 +7,7 @@
...
    *
    * @param {Object} routingConfig - possible URL`s hashes which user can type in and corresponding
    * class names of pages.
```
Comment (dkalinin) [inline app/router/index.js]: Please fix documentation for `onRouteChange` method.

File: app/service/fileservice/index.js [hunk 3]
```diff
@@ -7,20 +7,84 @@ export class FileService {
   /**
    * Sends a request to get all the files from the server.
    *
+   * @param {string} id - id of the folder, files from which should be requested.
    * @return {Promise<unknown>} server response.
    */
-  getFiles() {
-    return requestService.get();
+  getFiles(id) {
```
Comment (dkalinin) [inline app/service/fileservice/index.js]: Let's rename param to `folderId`.

File: app/service/fileservice/index.js [hunk 4]
```diff
@@ -7,20 +7,84 @@ export class FileService {
...
   }
 
   /**
    * Sends a request to remove file by its id.
    *
-   * @param {number} id - id of the file to be removed.
+   * @param {string} id - id of the file to be removed.
    * @return {Promise<unknown>} server response.
    */
   removeFile(id) {
```
Comment (dkalinin) [inline app/service/fileservice/index.js]: fileId.

Comment (dkalinin) [reply app/service/fileservice/index.js]: Check other methods.

File: app/storeconfiguration/action/index.js [hunk 5]
```diff
@@ -25,7 +25,7 @@ export default {
   // TODO:2019-08-29:bohdan.dudka: Remove items from get and fetch params.
   [FETCH_ITEMS_ACTION](context, payload) {
     context.commit(CHANGE_ARE_FILES_LOADING_MUTATION, true);
-    fileRequestService.getFiles()
+    fileRequestService.getFiles(payload)
```
Comment (dkalinin) [inline app/storeconfiguration/action/index.js]: Please remove payload to folderId.

**Pull request merged**